### PR TITLE
Compare round-tripped data for changes.

### PR DESCRIPTION
**Pull request by NeilFraser**
Clicking on an incompatible editor should not nullify the data.

### CODE REVIEW

**Comment by NeilFraser** [comment]
I ended up redesigning the system from scratch on paper, then working through the existing code to implement the diff.  Turned out that there was just one parameter that needed deleting.

File: static/code/editor.js [hunk 1]
```diff
@@ -440,15 +440,13 @@ Code.Editor.mostConfidentEditor = function() {
...
-Code.Editor.setSourceToAllEditors = function(src, isSaved) {
+Code.Editor.setSourceToAllEditors = function(src) {
   Code.Editor.uncreatedEditorSource = src;
   for (var editor of Code.Editor.editors) {
     editor.setSource(src);
-    if (isSaved) {
-      editor.lastSavedSource = editor.getSource();
-    }
+    // Round-trip version of the source.
+    editor.lastSavedSource = editor.getSource();
```
Comment (cpcallen) [inline static/code/editor.js]: O.K., I think I understand how this works, but maybe `lastSavedSource` should be renamed?  It seems to me it's *not* the last saved source, but rather the just the unmodified-by-the-user source.

Comment (NeilFraser) [reply static/code/editor.js]: Renamed to `unmodifiedSource`.

**Comment by NeilFraser** [comment]
It seems to behave exactly as one would want, with no trivial changes appearing unexpectedly.

Note that the grey save button is still live, and can still perform a save.  This allows one to force a trivial change that the editor doesn't deem worthy.  It also allows one to force a save and revert a value that was edited in another window (something I did all the time in Moo Canada, open two copies of the editor, and use one to execute reverts).

There is one slight wrinkle:  If one loads a function, and switches to the SVG editor, one gets a blank canvas as expected.  If one doesn't edit the SVG, but instead returns to the function editor, the original function is presented.  Regardless of tab switching, if the user hasn't touched anything the save button will never go blue.  This is all good.  However, if in the SVG editor one presses the grey save button, it will not save the blank SVG that is currently being presented, but rather resave the function.

This is a bit unexpected.  But after thinking about it for some time, I am of the opinion that this is the best available behaviour.  The reasons are:  a) There's rarely a reason to overwrite a value of a different type with a blank JSSP/SVG/function/etc.  b) The presence of a grey save button should consistently indicate that saving won't do anything (or at least not much).

**Comment by cpcallen** [comment]
That's totally satisfactory.  Go ahead and commit this.  I can push it to prod if you like.

**Comment by cpcallen** [comment]
Are you still tracking down some tab-clickhappiness weirdness?  It would be very nice to land this ASAP.  If you think it's not likely to be a major show-stopper, can it be dealt with later?

**Pull request merged**